main.go

```diff
@@ -166,3 +263,86 @@ func loadClient(kubeconfigPath string) (*k8s.Clientset, error) {
 	}
 	return k8s.NewForConfig(config)
 }
+
+func checkService(opts *Opts, client *k8s.Clientset) error {
+	logger.Infow("Function checkService called")
+	logger.Infow("Current service", "targetService", targetService)
+
+	kubectx, kubecancel := context.WithTimeout(context.Background(), time.Duration(10*time.Second))
+	defer kubecancel()
+	service, err := client.CoreV1().Services(opts.NameSpace).Get(kubectx, opts.ServiceName, metav1.GetOptions{})
+	if err != nil { // That means no matching service found
+		if targetService != nil { // This means a service was previously seen, and a forwarder should already be running.
+			logger.Infow("Service went away, killing forwarder")
+			killForwarder()
+			targetService = nil
+		}
+		return err
+	}
+	logger.Infow("Service gotten", "service", service)
+	serviceIP := service.Spec.ClusterIP
+	servicePort := service.Spec.Ports[0].Port
```

Comment:
Possible index out of bounds

Reply:
Looks like there must be at least one servicePort in a kubernetes service; trying to deploy one with empty spec: ports: [] gives:

```
kubectl apply -f kubernetes-audit-tailer-mod2.yaml
deployment.apps/kubernetes-audit-tailer configured
configmap/kubernetes-audit-tailer-config unchanged
The Service "kubernetes-audit-tailer" is invalid: spec.ports: Required value
```
Checking the length of the Ports slice(?) anyway, just to be sure.